op-service/txmgr/txmgr.go

```diff
@@ -421,6 +429,11 @@ func (m *SimpleTxManager) sendTx(ctx context.Context, tx *types.Transaction) (*t
 	defer ticker.Stop()
 
 	for {
+		// Check if there is a tx of incompatible type in the mempool that must clear or be
+		// cleared before this transaction can be sent.
+		if sendState.AlreadyReserved() {
+			return nil, ErrAlreadyReserved
+		}
```

Comment:
I think it makes sense to check this here, at the start of the for loop, because the `m.publishTx` call in `publishAndWait` is blocking and we're guaranteed to have recorded the error next time we enter the loop.

I see that we also have 
```
			if sendState.ShouldAbortImmediately() {
				m.txLogger(tx, false).Warn("Aborting transaction submission")
				return nil, errors.New("aborted transaction sending")
			}
```
inside the ticket block, which internally also checks for abort conditions from send errors.

I'm wondering if we could consolidate both to simplify the code and move that check up here. So we could change `ShouldAbortImmediately` to a method like `CriticalError() err` and then check here
```suggestion
		// Check if there is a tx of incompatible type in the mempool that must clear or be
		// cleared before this transaction can be sent.
		if err:=sendState.CriticalError(); err != nil {
			return nil, fmt.Errorf("critical send error: %w", err)
		}
```

Inside `CriticalError` we'd then check for all the current conditions inside `ShouldAbortImmediately` plus the already reserved error. This would have the added benefit that we would actually log why we aborted in those cases. Currently we have
```go
	// If we have exceeded the nonce too low count, abort
	if s.nonceTooLowCount >= s.safeAbortNonceTooLowCount ||
		// If we have not published a transaction in the allotted time, abort
		(s.successFullPublishCount == 0 && s.now().After(s.txInMempoolDeadline)) {
		return true
	}

	return false
```
which mixes both abort causes, too many nonce-too-low counts, and mempool deadline.

Reply:
done!